File: stepik-union/src/test/java/org/stepik/core/serialization/SampleConverterTest.java
package org.stepik.core.serialization;

import org.junit.Before;
import org.junit.Test;
import org.stepik.api.objects.steps.Sample;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;

/**
 * @author meanmail
 */
public class SampleConverterTest {
    private SampleConverter sampleConverter;

    @Before
    public void setUp() {
        sampleConverter = new SampleConverter();
    }

    @Test
    public void toStringTest() throws Exception {
        Sample sample = new Sample();
        sample.setInput("1 2 3");
        sample.setOutput("123");

        String sampleString = sampleConverter.toString(sample);

        assertEquals("<input>1 2 3</input><output>123</output>", sampleString);
    }

    @Test
    public void fromStringTest() throws Exception {
        String string = "<input>1 2 3</input><output>123</output>";

        Object sample = sampleConverter.fromString(string);

        Sample expected = new Sample();
        expected.setInput("1 2 3");
        expected.setOutput("123");

        assertEquals(expected, sample);
    }

    @Test
    public void canConvert() throws Exception {
        boolean value = sampleConverter.canConvert(Sample.class);
        assertTrue(value);
    }

}

File: stepik-core/src/test/kotlin/org/stepik/core/serialization/SampleConverterTest.kt
package org.stepik.core.serialization

import org.junit.Assert.assertEquals
import org.junit.Assert.assertTrue
import org.junit.Before
import org.junit.Test
import org.stepik.api.objects.steps.Sample


class SampleConverterTest {
    private var sampleConverter: SampleConverter? = null

    @Before
    fun setUp() {
        sampleConverter = SampleConverter()
    }

    @Test
    fun toStringTest() {
        val sample = Sample()
        sample.input = "1 2 3"
        sample.output = "123"

        val sampleString = sampleConverter!!.toString(sample)

        assertEquals("<input>1 2 3</input><output>123</output>", sampleString)
    }

    @Test
    fun fromStringTest() {
        val string = "<input>1 2 3</input><output>123</output>"

        val sample = sampleConverter!!.fromString(string)

        val expected = Sample()
        expected.input = "1 2 3"
        expected.output = "123"

        assertEquals(expected, sample)
    }

    @Test
    fun canConvert() {
        val value = sampleConverter!!.canConvert(Sample::class.java)
        assertTrue(value)
    }

}
